### PR TITLE
Enable specifying items on Template construction

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
This PR implements the FR https://github.com/holoviz/panel/issues/1693.

I.e. it provides support for providing header, sidebar, main and modal item(s) on construction.

The implementation is ready for review @philippjfr . The PR includes the relevant tests. But you can also manually play around with the example below.

## Example

You can manually test via

```python
from panel.template import ReactTemplate, VanillaTemplate
from panel.pane import Markdown

# item1 = Markdown("Hello Header")
# item2 = Markdown("Hello Sidebar")
# item3 = Markdown("Hello Main")

item1 = [Markdown("Hello Header"), Markdown("Hello2")]
item2 = [Markdown("Hello Sidebar"), Markdown("Hello2")]
item3 = [Markdown("Hello Main"), Markdown("Hello2")]
# VanillaTemplate(title="VanillaTemplate", header=item1, sidebar=item2, main=item3).servable()
ReactTemplate(title="ReactTemplate", header=item1, sidebar=item2, main=item3).servable()
```

which looks like

image

### CODE REVIEW

File: panel/template/base.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from __future__ import absolute_import, division, unicode_literals
 
 import os
+from panel.layout.grid import GridSpec
```
Comment (philippjfr) [inline panel/template/base.py:8]: Imports should be relative and be ordered correctly, i.e. you should use `from ..layout.grid import GridSpec` and then move this import into the list of internal imports.

File: panel/template/base.py [hunk 2]
```diff
@@ -570,6 +579,30 @@ def _get_favicon_type(favicon):
...
+        if class_==GridSpec:
+            grid = GridSpec(ncols=12)
+            for index, item in enumerate(value):
+                grid[index, :]=item
+            return grid
+
+        return value
+
+
+
```
Comment (philippjfr) [inline panel/template/base.py:605]: Too much space.

File: panel/template/base.py [hunk 3]
```diff
@@ -570,6 +579,30 @@ def _get_favicon_type(favicon):
...
+            return value
+        if isinstance(value, tuple):
+            value = [*value]
+        elif not isinstance(value, list):
+            value = [value]
+
+        # Important to fx. convert @param.depends functions
+        value = [_panel(item) for item in value]
+
+        if class_==ListLike:
```
Comment (philippjfr) [inline panel/template/base.py:594]:
```suggestion
        if class_ is ListLike:
```

File: panel/template/base.py [hunk 4]
```diff
@@ -570,6 +579,30 @@ def _get_favicon_type(favicon):
...
+            return ListLike(objects=value)
+        if class_==GridSpec:
```
Comment (philippjfr) [inline panel/template/base.py:596]:
```suggestion
        if class_ is GridSpec:
```

File: panel/template/base.py [hunk 5]
```diff
@@ -570,6 +579,30 @@ def _get_favicon_type(favicon):
...
+            grid = GridSpec(ncols=12)
+            for index, item in enumerate(value):
+                grid[index, :]=item
```
Comment (philippjfr) [inline panel/template/base.py:599]: Suppose this is reasonable, but it is *a bit* weird.

**Comment by philippjfr** [comment]
Mostly nitpicky comments. If you fix it up this by about noon this can be the last commit before a release.

File: panel/template/base.py (hunk 5 above)
Comment (MarcSkovMadsen) [reply panel/template/base.py:599]: This comment I don't understand. What is weird?

If you are referring to the `ncols=12` I've taken that from the ReactTemplate constructor

image

**Comment by MarcSkovMadsen** [comment]
Thanks for the review @philippjfr . You are a man of **style** :-) That is great. 

I've pushed fixes based on your comments. Except the *weird* comment which I do not understand.

As a side note I need to get a good QA flow for working with Panel. My normal flow is different. It's automated based on isort, autoflake8, black, pylint, mypy and pytest in that order. So most things are automated away and I don't have to spend mental band width on it. So I need to learn to do more manual things here.

File: panel/template/base.py (hunk 5 above)
Comment (philippjfr) [reply panel/template/base.py:598]: Iterating over the list and assigning each to a row is okay behavior but it's certainly not the only behavior that makes sense. Anyway I'm fine with it.

**Comment by philippjfr** [comment]
I think we should definitely enforce a few more code quality checks but for now let's release 🥳

**Pull request merged**